src/frontend-samples/read-settings-sample/ReadSettingsWidget.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { ChangeEvent, useCallback, useEffect } from "react";
+import { useActiveIModelConnection } from "@bentley/ui-framework";
+import { AbstractWidgetProps, StagePanelLocation, StagePanelSection, UiItemsProvider, WidgetState } from "@bentley/ui-abstract";
+import { Button, DisabledText, Select, SmallText, Spinner, SpinnerSize, Textarea } from "@bentley/ui-core";
+import { SettingsResult, SettingsStatus } from "@bentley/product-settings-client";
+import ReadSettingsApi from "./ReadSettingsApi";
+import "./ReadSettings.scss";
+
+const ReadSettingsWidget: React.FunctionComponent = () => {
+  const settingsKeys = ["Json_Data", "Arbitrary_Text", "CSV_Data"];
+
+  const iModelConnection = useActiveIModelConnection();
+  const [settingKey, setSettingKey] = React.useState<string>(settingsKeys[0]);
+  const [settingValue, setSettingValue] = React.useState<string>();
+  const [settingResult, setSettingResult] = React.useState<SettingsResult>();
+  const [saveInProgress, setSaveInProgress] = React.useState<boolean>(false);
+  const [readUpdate, setReadUpdate] = React.useState<boolean>(true);
+  const [saveUpdate, setSaveUpdate] = React.useState<boolean>(false);
+
+  const _parseSettingsValue = useCallback((name: string, value: string) => {
+    let _value;
+    switch (name) {
+      case settingsKeys[0]:
+        _value = JSON.stringify(value || "", null, "   ");
+        break;
+      default:
+        _value = value || "";
+    }
+    return _value;
+  }, [settingsKeys]);
+
+  /** iModelConnection react state is very sensitive and updates a lot, so */
```

Comment:
... so check readUpdate flag?

This seems to be an example of the exhaustive-deps being annoying?

Reply:
Added further comments. Yeah the dependency is very finicky.